src/DependencyInjection/EmagCacheExtension.php

```diff
@@ -31,6 +35,19 @@ public function prepend(ContainerBuilder $container)
         if (!$provider->implementsInterface(CacheItemPoolInterface::class)) {
             throw new CacheException(sprintf('You\'ve referenced a service "%s" that can not be used for caching!', $config['provider']));
         }
+
+        if (!$config['expression_language']) {
+            return;
+        }
+
+        if (!class_exists('Symfony\Component\ExpressionLanguage\ExpressionLanguage')) {
+            throw new CacheException('Unable to use expressions as the Symfony ExpressionLanguage component is not installed.');
+        }
+
+        $expressionLanguage = new \ReflectionClass($container->getDefinition($config['expression_language'])->getClass());
+        if ($expressionLanguage->getName() !== ExpressionLanguage::class) {
+            throw new CacheException(sprintf('You must provide a valid Expression Language service'));
```

Comment:
Too exact class match. You should be able to provide anything that extends expressionLanguage.